docs/azure_arc_jumpstart/azure_arc_data/microk8s/azure/arm_template/dc_vanilla/_index.md

```diff
@@ -0,0 +1,206 @@
+title: "Data Controller ARM Template"
+linkTitle: "Data Controller ARM Template"
+weight: 1
+description: >
+---
+
+## Deploy a vanilla Azure Arc Data Controller in directly connected mode on a Microk8s Kubernetes cluster in an Azure VM using ARM Templates
+
+The following README will guide you on how to deploy a "Ready to Go" environment so you can start using [Azure Arc enabled data services](https://docs.microsoft.com/en-us/azure/azure-arc/data/overview) deployed on a single-node [Microk8s](https://microk8s.io/) Kubernetes cluster.
+
+By the end of this guide, you will have a Microk8s Kubernetes cluster deployed with an Azure Arc Data Controller and a Microsoft Windows Server 2019 (Datacenter) Azure Client VM, installed & pre-configured with all the required tools needed to work with Azure Arc enabled data services:
+
+![Deployed Architecture](./01.png)
+
+> **Note: Currently, Azure Arc enabled data services is in [public preview](https://docs.microsoft.com/en-us/azure/azure-arc/data/release-notes)**.
+
+## Prerequisites
+
+- Clone the Azure Arc Jumpstart repository
+
+  ```shell
+  git clone https://github.com/microsoft/azure_arc.git
+  ```
+
+- [Install or update Azure CLI to version 2.15.0 and above](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest). Use the below command to check your current installed version.
+
+  ```shell
+  az --version
+  ```
+
+- [Generate SSH Key](https://docs.microsoft.com/en-us/azure/virtual-machines/linux/create-ssh-keys-detailed) (or use existing ssh key).
+
+- Create Azure service principal (SP)
+
+  To be able to complete the scenario and its related automation, Azure service principal assigned with the “Contributor” role is required. To create it, login to your Azure account run the below command (this can also be done in [Azure Cloud Shell](https://shell.azure.com/)).
+
+  ```shell
+  az login
+  az ad sp create-for-rbac -n "<Unique SP Name>" --role contributor
+  ```
+
+  For example:
+
+  ```shell
+  az ad sp create-for-rbac -n "http://AzureArcData" --role contributor
+  ```
+
+  Output should look like this:
+
+  ```json
+  {
+    "appId": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX",
+    "displayName": "AzureArcData",
+    "name": "http://AzureArcData",
+    "password": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX",
+    "tenant": "XXXXXXXXXXXXXXXXXXXXXXXXXXXX"
+  }
+  ```
+
+  > **Note: It is optional, but highly recommended, to scope the SP to a specific [Azure subscription and resource group](https://docs.microsoft.com/en-us/cli/azure/ad/sp?view=azure-cli-latest).**
+
+- Enable subscription for the _Microsoft.AzureArcData_ resource provider for Azure Arc enabled data services. Registration is an asynchronous process, and registration may take approximately 10 minutes.
+
+  ```shell
+  az provider register --namespace Microsoft.AzureArcData
+  ```
+
+  You can monitor the registration process with the following commands:
+
+  ```shell
+  az provider show -n Microsoft.AzureArcData -o table
+  ```
```

Comment:
This should go inside to automation. See [here](https://github.com/microsoft/azure_arc/blob/main/azure_arc_data_jumpstart/cluster_api/capi_azure/arm_template/artifacts/DataServicesLogonScript.ps1#L38)

Reply:
Removed from markdown, as it's already inside [`DataServicesLogonScript.ps1`](https://github.com/microsoft/azure_arc/blob/60497dd3f3b590b97c2ca52b9f6849e989d9c21e/azure_arc_data_jumpstart/microk8s/azure/arm_template/artifacts/DataServicesLogonScript.ps1) to your point.

Only point I want to highlight is, `az provider register` will require the Service Principal the user creates to have **Subscription** level [**Owner**/**Contributor**](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/resource-providers-and-types#register-resource-provider) permissions to register the provider.

So for example, say someone runs this scenario the very first time without having already registering `AzureArcData` before, and the SP they created is only RG scoped - in this situation when the script runs inside the Client VM, the SP won't be able to register the provider.

(But if this happens in this scenario, the user can run the registrations manually using their own creds against the Subscription).